src/node_crypto.cc

```diff
@@ -1681,10 +1697,49 @@ static Local<Object> X509ToObject(Environment* env, X509* cert) {
         reinterpret_cast<unsigned char*>(Buffer::Data(pubbuff));
     i2d_RSA_PUBKEY(rsa.get(), &pubserialized);
     info->Set(env->context(), env->pubkey_string(), pubbuff).FromJust();
+  } else if (ec) {
+    const EC_GROUP* group = EC_KEY_get0_group(ec.get());
+    if (group) {
+      int ecbits = EC_GROUP_order_bits(group);
+      if (ecbits > 0) {
+        info->Set(context, env->ecbits_string(),
+                  Integer::New(env->isolate(), ecbits)).FromJust();
+      }
+    }
+
+    unsigned char* pub = nullptr;
+    size_t publen = EC_KEY_key2buf(ec.get(), EC_KEY_get_conv_form(ec.get()),
+                                   &pub, nullptr);
+    if (publen) {
```

Comment:
`publen > 0` (I'll stop pointing this out, just use explicit relation operators for values that aren't logically bool.)